src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -74,13 +74,13 @@ public Stage getPrimaryStage() {
      */
     void fillInnerParts() {
 
-        taskListPanel = new TaskListPanel(logic.getVisualList());
+        taskListPanel = new TaskListPanel(logic.getTaskList());
```

Comment:
This will work if the different lists are static. However, when using a method like find or sort, this will not work as the task list should not be the one changing, but the visual list changing. Therefore this change will not be reflected if we are pointing at the different lists.

Reply:
If the individual lists change, we may not have a solution to retrieve back the original list. (Find and sort should change a viewing list while maintaining the original list)